### ISSUE TITLE
Best approach for underwater SfM

### ISSUE THREAD

**Issue by KevinCain** [openMVG/openMVG#553]
I gather that OpenMVG requires basic camera intrinsic parameters (focal length in mm, image sensor aspect) while the remaining basic intrinsic parameter, principal point, is deduced.

When working with underwater photos, the air/water interfaces acts like a secondary lens, which has an effect on the intrinsics.  Should I attempt to correct for the input focal length, or will the SfM process optimize the radial distortion terms in the default camera model in order to partially account for the change in the index of refraction?

@pmoulon has noted that some users have successfully obtained good results with underwater, close-range photography (e.g.:  strong parallax, no weakly supported angles), so the 'best practices' here may already be known for OpenMVG.

Thanks for any advice or questions on this topic!

**Comment by pmoulon** [comment]
Perhaps @lucaarcteam & @alearcteam will be able to provide some feedback since they have tested it on field.

**Comment by lucaarcteam** [comment]
Hi @KevinCain ,
here some report of what we did by now:

http://arc-team-open-research.blogspot.it/2013/03/extreme-sfm-underwater-archaeology.html

If you want to know a little bit more about the project, here are two other post:

http://arc-team-open-research.blogspot.it/2011/07/lake-tovel-red-lake.html

and

http://arc-team-open-research.blogspot.it/2014/08/the-underwater-forest-of-lake-tovel.html

Soon we will start again diving and trying other 3D techniques. Currently we are developing/building an open hardware rover to perform 3D underwater documentation and experiment different sensors. I hope to write soon some post about it in ATOR (http://arc-team-open-research.blogspot.it/). If you need more info, just reply here. Have a nice day!

**Comment by KevinCain** [comment]
Thank you @lucaarcteam for sharing some of your inspiring work!  The engineers from OpenROV were kind enough to come to our studio a few years back, they (and I) share you enthusiasm for making open hardware for underwater sampling.

I saw your posts, thank you, but perhaps I missed any notes on your use of OpenMVG.  My main question in this thread is how you defined your camera.  Did you use a standard entry in the OpenMVG database for your camera, or add your own?  If you added your own, did you use the published value of your camera's focal length, or modify it to account for the 'secondary lens' effect I briefly sketched above?

**Comment by lucaarcteam** [comment]
Sorry @KevinCain I just linked the post without checking if there were more info (they are pretty old). When we did that test, openmvg was not yet developed by Pierre and we used the software he did before (Python Photogrammetry Toolbox), which was a python wrapper for Bundler, CMVS and PMVS2 (Alessandro wrote a fast GUI in python for it). We just extract some photograms from the video (done with a GoPro) you see in the post (if I remember well with imagemagick); than we used them with PPT in order to get the 3d pointcloud. We did not use special camera parameters and the result was pretty good for us (the goal is to do an underwater 3D GIS for the trees of that lake).
Currently we use more than one software for 3D. In short:

1. openmvg, for most of the projects (especially for documentation from the ground for architecture and archaeological excavations); here (http://arc-team-open-research.blogspot.it/2016/02/arc-team-2015-iran-mission-video.html) an example
2. MicMac, in general for aerial archaeology project with our open hardware drones; in these post (http://arc-team-open-research.blogspot.it/2014/05/micmac-and-ppt-two-floss-solutions-for.html) you see how many points MicMac can do in these kind of projects
3. PPT; we still use it because it gave us the better results in particular situation, like to document underground environment (e.g. http://arc-team-open-research.blogspot.it/2013/03/sfm-for-underground-documentation.html)

For the rover we are planning to use slam algorithms to map in 3d in real-time. We did some test on the ground and they work pretty good for what we have to do underwater. The plan is SLAM for 3d general map and openmvg for object 3d documentation, but we plan to try to develop also an open hardware sonar. By now we are building another model of rover (not the openrov), because we need some particular devices, so we started from the beginning.
